neural_modelling/src/neuron/implementations/neuron_impl_semd.h

```diff
@@ -0,0 +1,219 @@
+#ifndef _NEURON_IMPL_SEMD_H_
+#define _NEURON_IMPL_SEMD_H_
+
+#include "neuron_impl.h"
+
+// Includes for model parts used in this implementation
+#include <neuron/models/neuron_model_lif_impl.h>
+#include <neuron/threshold_types/threshold_type_static.h>
+#include <neuron/synapse_types/synapse_types_exponential_impl.h>
+
+// Further includes
+#include <common/out_spikes.h>
+#include <recording.h>
+#include <debug.h>
+#include <string.h>
+
+#define V_RECORDING_INDEX 0
+#define GSYN_EXCITATORY_RECORDING_INDEX 1
+#define GSYN_INHIBITORY_RECORDING_INDEX 2
+
+typedef struct input_type_current_semd_t {
+
```

Comment:
I think it's fine to do this here, but it seems a bit strange to do it here rather than actually in the input_type directory.  I understand, though, that it's a bit weird to add functions to the standard input_type that will never be used by it... so I'm happy with this the way it is - just thought I would mention this here in case any more thought is required.

Reply:
The SEMD model is a little bit odd certainly, in that it almost, but doesn't quite, fit in to the standard neuron model.  As you say though, we can't really add a function each time someone wants something different from the interface, as this would be hard to keep up with!

The SEMD implementation ended up being custom at the C level, but actually standard at the Python level.  As I say, this is a bit odd but it does work!

sPyNNaker8NewModelTemplate now contains an example where this is not the case.  Hopefully that helps to show what a completely non-standard model looks like...